nixos/modules/services/web-apps/crabfit.nix

```
@@ -0,0 +1,171 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    literalExpression
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    ;
+
+  inherit (lib.types)
+    attrsOf
+    package
+    port
+    str
+    ;
+
+  cfg = config.services.crabfit;
+in
+
+{
+  options.services.crabfit = {
+    enable = mkEnableOption "Crab Fit, a meeting scheduler based on peoples' availability";
+
+    frontend = {
+      package = mkPackageOption pkgs "crabfit-frontend" { };
+
+      finalDrv = mkOption {
+        readOnly = true;
+        type = package;
+        default = cfg.package.override {
+          api_url = "https://${cfg.api.host}";
+          frontend_url = cfg.frontend.host;
+        };
+
+        defaultText = literalExpression ''
+          cfg.package.override {
+            api_url = "https://''${cfg.api.host}";
+            frontend_url = cfg.frontend.host;
+          };
+        '';
+
+        description = ''
+          The patched frontend, using the correct urls for the API and frontend.
+        '';
+      };
```

Why are you splitting this out into two options instead of doing something like:
```suggestion
      package = mkOption {
        readOnly = true;
        type = package;
        default = cfg.package.override {
          api_url = "https://${cfg.api.host}";
          frontend_url = cfg.frontend.host;
        };

        defaultText = literalExpression ''
          cfg.package.override {
            api_url = "https://''${cfg.api.host}";
            frontend_url = cfg.frontend.host;
          };
        '';

        description = ''
          The patched frontend, using the correct urls for the API and frontend.
        '';
      };
```

---

How do you override `package` in your proposal?

---

something like:
```nix
config.services.crabfit.frontend.package = (pkgs.crabfit-frontend.override {
          api_url = "https://${cfg.api.host}";
          frontend_url = cfg.frontend.host;
        });
```

but I'm fine with leaving it as is until https://github.com/NixOS/nixpkgs/pull/300917#discussion_r1552093453 is resolved.

---

But.. you write a read only option :P.

---

edited the suggestion :see_no_evil: 
but as I said, I'm fine with having it as is until the other comment is resolved.